### DISCUSSION TITLE
[Quest] Update Sync mechanism to Rely on Composition or Sync by  Strategy

### DISCUSSION THREAD

**Discussion by f-odhiambo** [opensrp/fhircore#1642] [Ideas]
We may need  to update the Sync mechanisms to

Non Patient data to be synced via Composition 
1. Binary 
2. Questionnaire 
3. Structure Maps

Patient Data to be synced via Sync by Strategy 
1. Condition 
2. Observations
4. Encounters 
5. PlanDefinitions
6. CarePlans
7. ServiceRequest 
8. QuestionnaireResponse

Sync by Strategy defined here https://github.com/opensrp/fhircore/discussions/746 

**Pre-requisite** 
1. Documentation of what composition resource would hold 
2. What data will be synced/tagged  based on the user assignment 
3. How can we give a user feedback on what is being synced / configured  - do we need a new progress bar (UX to be discussed with design team)

**Comment by pld** [comment]
There are further details on this we discussed today, I'm going to convert this to a discussion until this is ready for assignment.

Let's call the non-patient data the config data. We also want to split the config  data into blocking and non-blocking, non-blocking data for sync we can background. So the process looks something like this with time from top to bottom:

1. Sync blocking config data
    - Composition
    - Linked Binaries
    - Linked Questionnaires, linked Structure maps

[app is now usable]

2. Background sync non-blocking config data
   - Measures
   - Libraries needed for measures
   - [Anything else?]

[all configs sync, data not synced]

3. Sync patient data
    - Show progress bar on Registers, @rowo we need design teams input on whether we want to do this and wait until sync is complete, or also show the patient data we have so far in Register as it comes w/some notice it's still loading, or something else
